Make attributes in AttrDict properly writable

rbscholtus opened the pull request:
`AttrDict(dict)` is a very convenient class for accessing confuse dictionaries using attribute style notation, e.g. `valid.paths.directory`. 

However, you cannot use attribute style notation to (over)write attributes. For example `valid.paths.directory = "C:/"` will not work as expected. It will create a new attribute `directory` in `valid.paths`. This is because the set statement accesses `directory` in subclass `AttrDict`, not the `dict` base class. To overwrite the existing attribute, you have to write `valid['paths']['directory'] = "C:/"` instead :(.

This pull request solves that problem by writing attributes to the base class.

sampsyo commented:
Seems like a good idea! Do you think you could add a quick entry to the changelog? https://github.com/beetbox/confuse/blob/master/docs/usage.rst#changelog

rbscholtus commented:
Yes definitely. Is there anything else, like test cases or documentation?

Barend

On Thu, 19 Nov 2020, 20:07 Adrian Sampson, <notifications@github.com> wrote:

> Seems like a good idea! Do you think you could add a quick entry to the
> changelog?
> https://github.com/beetbox/confuse/blob/master/docs/usage.rst#changelog
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/beetbox/confuse/pull/120#issuecomment-730362975>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AADGXQM6AU4R2R2EDRKBPUTSQUKAXANCNFSM4T3KBGUQ>
> .
>

sampsyo commented:
Both would be awesome if you don't mind looking into them! However:

- I'm not entirely sure where the docs would go for this—if there's a place where we currently discuss "dot" access to dictionaries, then we can add it there. If not, don't worry about it.
- I actually don't think we have specific tests for the AttrDict functionality. So like with the docs, if we don't already have scaffolding for this, don't worry about inventing it.

rbscholtus commented:
I updated the pull request (4 files instead of 1 file.)

* I added a changelog for v1.5.0. Is this the right version? (ie not 1.4.1)
* I found further references to the version nr (confuse\confuse\__init__.py
and confuse\docs\conf.py) but they have been out-of-date as it stands (ie
v1.3.0). Do you mind checking this for me?
* The docs make no specific reference to dot notation.
* No test related to AttrDict or dot notation.

How does this look?

On Thu, 19 Nov 2020 at 20:33, Adrian Sampson <notifications@github.com>
wrote:

> Both would be awesome if you don't mind looking into them! However:
>
>    - I'm not entirely sure where the docs would go for this—if there's a
>    place where we currently discuss "dot" access to dictionaries, then we can
>    add it there. If not, don't worry about it.
>    - I actually don't think we have specific tests for the AttrDict
>    functionality. So like with the docs, if we don't already have scaffolding
>    for this, don't worry about inventing it.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/beetbox/confuse/pull/120#issuecomment-730377126>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AADGXQJQXEPALGART7G6QNTSQUNDRANCNFSM4T3KBGUQ>
> .
>


-- 

Barend
Certified ScrumMaster

sampsyo commented:
This all looks perfect! Thank you!!

rbscholtus commented:
When do you reckon it'll be available on pypi.org?
( I need it heheheh)
Many thanks professor.
Barend


On Sat, 21 Nov 2020 at 06:08, Adrian Sampson <notifications@github.com>
wrote:

> Merged #120 <https://github.com/beetbox/confuse/pull/120> into master.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/beetbox/confuse/pull/120#event-4023095718>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AADGXQPTIDBH3PVXYRJIJNLSQ3ZGHANCNFSM4T3KBGUQ>
> .
>


-- 

Barend
Certified ScrumMaster

sampsyo commented:
Published! It turned out we hadn't published 1.4.0 yet, so I folded the change into that.

rbscholtus commented:
1.4.0, 3 minutes ago ;)
Have a great day prof!

On Sun, 22 Nov 2020, 01:29 Adrian Sampson, <notifications@github.com> wrote:

> Published! It turned out we hadn't published 1.4.0 yet, so I folded the
> change into that.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/beetbox/confuse/pull/120#issuecomment-731617560>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AADGXQIKGWYCPV75DBL2KVLSRABG3ANCNFSM4T3KBGUQ>
> .
>
